tensorflow/core/grappler/optimizers/auto_mixed_precision.cc

```diff
@@ -1159,6 +1162,7 @@ std::pair<int, int> GetDeviceGPUArch(
 
 bool AutoMixedPrecisionImpl::IsOnSuitableGPUArch(const NodeDef& node) const {
   return GetDeviceGPUArch(virtual_placer_.get_device(node)) >= kMinGPUArch;
```

Comment:
Does `GetDeviceGPUArch` work correctly for AMD GPUs? 

Also I think for AMD, even though the numbers are in increasing order (gfx803, gfx900, gfx906, etc), the recommendation is to not use number comparison as a means for testing whether a given feature is supported. Because it is possible that 900 and 908 support some feature but 906 does not! 
So the routines for doing hardware specific feature checks for AMD should be something like 

```
...
return HasEnhancedFP16ComputeSupport(GetDeviceGPUArch(virtual_placer_.get_device(node)));
...
bool HasEnhancedFP16ComputeSupport(std::pair<int, int> gpu_arch)  
// data-type for gpu_arch arg can be anything you want...we are not limited to using int pair
{
// this routine should return true for gpu_arch that has enhanced fp16 support, false otherwise
}
```

Reply:
Ok. I'll look into a more sophisticated approach for this.

Reply:
> Because it is possible that 900 and 908 support some feature but 906 does not!

that was meant as an example...not specific to fp16 support. For enhanced fp16 support (i.e. where we have fp16 compute support in HW, I believe gfx908 has it, gfx900 does not have it, and I do know for sure about the rest)

@mvermeulen, is there some wiki-page and / or document which lists which HW features are present in which gfx arch?

Reply:
@deven-amp Peng sent me a doc with the features. I misread what gfx900 had, see my latest commit. 906 actually has the FP16 it appears, whereas 900 has "Packed" FP16.